client/state-db/src/lib.rs

```diff
@@ -98,12 +115,22 @@ impl<
 }
 
 /// Backend database trait. Read-only.
-pub trait MetaDb {
+pub trait MetaDbRead {
 	type Error: fmt::Debug;
 
 	/// Get meta value, such as the journal.
 	fn get_meta(&self, key: &[u8]) -> Result<Option<DBValue>, Self::Error>;
 }
+pub trait MetaDbWrite {
```

Comment:
```suggestion

pub trait MetaDbWrite {
```

nitpick, feel free to ignore